builtin/rebase.c

```diff
@@ -287,6 +301,10 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 			die(_("Could not resolve HEAD to a revision"));
 	}
 
+	/* If a hook exists, give it a chance to interrupt*/
+	run_hook_le(NULL, options.upstream_arg
+			, options.ok_to_skip_pre_rebase, NULL);
```

Comment:
Please do not use a leading `,` in the formatting, no existing source code in Git uses this convention.

Besides, at [this point](https://github.com/prertik/git/blob/ae91dbb2556d7ea9f4e205fbf7847fa8579c64b4/TODO-rebase.sh#L700), `"$@"` (i.e. the remaining command-line arguments) are not "ok_to_skip_pre_rebase", but the branch name to switch to (if any has been provided, otherwise there is no remaining command-line argument).

Reply:
Furthermore, I suspect that this code was probably too confusing? https://github.com/prertik/git/blob/ae91dbb2556d7ea9f4e205fbf7847fa8579c64b4/TODO-rebase.sh#L232-L233

What it does is first test whether the flag was set indicating that it is okay to skip the pre-rebase hook. If it is okay, the hook is not executed! This would translate into this snippet in C, though:

```
if (options.ok_to_skip_pre_rebase)
    run_hook_le(...)
```

The second part tests whether the hook in question is executable (and exists, in the first place). In C, the `run_hook_le()` function already does that for you, no need to convert that.